common/src/voxel_math.rs

```diff
@@ -89,3 +107,154 @@ impl IndexMut<CoordAxis> for Coords {
         self.0.index_mut(coord_axis as usize)
     }
 }
+
+/// Represents one of the six main directions within a chunk: positive or negative x, y, and z.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct ChunkDirection {
+    pub axis: CoordAxis,
+    pub sign: CoordSign,
+}
+
+impl ChunkDirection {
+    pub const PLUS_X: Self = ChunkDirection {
+        axis: CoordAxis::X,
+        sign: CoordSign::Plus,
+    };
+    pub const PLUS_Y: Self = ChunkDirection {
+        axis: CoordAxis::Y,
+        sign: CoordSign::Plus,
+    };
+    pub const PLUS_Z: Self = ChunkDirection {
+        axis: CoordAxis::Z,
+        sign: CoordSign::Plus,
+    };
+    pub const MINUS_X: Self = ChunkDirection {
+        axis: CoordAxis::X,
+        sign: CoordSign::Minus,
+    };
+    pub const MINUS_Y: Self = ChunkDirection {
+        axis: CoordAxis::Y,
+        sign: CoordSign::Minus,
+    };
+    pub const MINUS_Z: Self = ChunkDirection {
+        axis: CoordAxis::Z,
+        sign: CoordSign::Minus,
+    };
+
+    pub fn iter() -> impl ExactSizeIterator<Item = ChunkDirection> {
+        [
+            Self::PLUS_X,
+            Self::PLUS_Y,
+            Self::PLUS_Z,
+            Self::MINUS_X,
+            Self::MINUS_Y,
+            Self::MINUS_Z,
+        ]
+        .iter()
+        .copied()
```

Comment:
style nit: as of a semi-recent Rust update, we can use `into_iter()` directly on an array